starting known issue for search (#621)
open-cluster-management/rhacm-docs: release_notes/known_issues.md

For technical preview, documentation links were removed from the console, but a few might still be exposed. Any links to the documentation for preview are temporarily not updated with the correct links.

> ### Search component memory leak

Reviewer: The title should describe the symptom the user might see. Can we change to:  "Search unavailable or missing data for a brief period"

Reviewer: Should we add Search to the table of contents?

Author: @jlpadilla are you referring to the summary.md file when you mention "table of contents? Or are you referring to the TOC on the known issues page?

Reviewer: The TOC at the top of this file.

Author: Oh no, the intention is to separate according to our top-level pages/components. It is due to the conversion to ASCII.

> <!--1.0.0:2228-->

Reviewer: Should this comment have the actual bug #1918 ? I think that will make it easier to update once the bug is fixed.

> The memory used by the search data store is being used at a fast rate and causes pods to restart. Search becomes unavailable or the data is out of sync with the hub cluster. The memory leak accelerates when more managed clusters are added. 

Reviewer: Actually, the memory grows *slowly* over time, because unused memory isn't being released.  The result is that although slowly, the memory is always growing, never shrinking, so eventually we will reach the limit.
```suggestion
The memory used by the search data store grows slowly over time, causing the pod to restart when reaching the limit. Search becomes unavailable or the data is out of sync with the managed clusters. The memory leak accelerates when more managed clusters are added.
```

You might encounter the following issues:

> * Search is unable to display some cluster status

Reviewer: - Add main symptom, which is that search becomes out of sync.
- These look like complete sentences, do you need to end with a period?

```suggestion
* Search becomes unavailable or data is out of sync for a brief period
* Search is unable to display some cluster status
```

* Search is unable to display results for newly created applications

> You can increase the memory limit in the `search-pod-xxxxx-redisgraph` deployment to reduce the frequency of pod restarts. Complete the following steps to update the deployment:

Reviewer: Might help to stress that this is a *mitigation*, not the solution to the problem.
```suggestion
To mitigate this problem, you can increase the memory limit in the `search-pod-xxxxx-redisgraph` deployment to reduce the frequency of pod restarts. Complete the following steps to update the deployment:
```

* Run the following command to increase the memory limit in the `search-pod-xxxxx-redisgraph` deployment from the command line interface (CLI):

   ```
   oc patch deployment search-prod-xxxxx-redisgraph -n open-cluster-management -p '{"spec": {"template": {"spec": {"containers":[{"name":"redisgraph","resources": {"limits": {"memory": "4Gi"}}}]}}}}'
   ```

* Update your `search-pod-xxxxx-redisgraph` deployment from the console:

  1. Log in to the Red Hat Advanced Cluster Management for Kubernetes cluster.
  2. Navigate to the _Search_ page and enter `search-prod-xxxxx-redisgraph` deployment. 

>   3. Update the `containers.resources.limit.memory` parameter and increase the memory value. **Note**: Your maximum memory is restricted by either your quota, policies, or physical limits of the nodes on your cluster.

Reviewer: The note applies to both methods, so let's separate it from step 3.
```suggestion
  3. Update the `containers.resources.limit.memory` parameter and increase the memory value.
  
**Note**: Your maximum memory is restricted by either your quota, policies, or physical limits of the nodes on your cluster.
```
